examples/fleet.yaml

```diff
@@ -15,25 +14,40 @@ spec:
       os:
         image: quay.io/redhat/rhde:9.2
       config:
-        - name: base
-          configType: GitConfigProviderSpec
-          gitRef:
-            repository: flightctl-demos
-            targetRevision: main
-            path: /demos/basic-nginx-demo/configuration
-        - name: motd-update
-          configType: InlineConfigProviderSpec
-          inline:
-            ignition:
-              version: 3.4.0
-            storage:
-              files:
-                - contents:
-                    source: >-
-                      data:,This%20system%20is%20managed%20by%20flightctl.%0A
-                  mode: 422
-                  overwrite: true
-                  path: "/etc/motd"
+        source: 
```

Comment:
why changing the name from config to source?

Reply:
It is technically config.source as config now is being extended. I am open to other names but since it now refers to the source of the config via provider on the fleet level I went with it.

Reply:
I believe this feature is config hooks because the hook is into the application of the configuration files to disk.